db/commitlog/commitlog.cc

```diff
@@ -1303,6 +1385,36 @@ db::commitlog::segment_manager::segment_manager(config c)
     }
 }
 
+void db::commitlog::segment_manager::send(message m) {
+    _message_queue.push(std::move(m));
+}
+
+void db::commitlog::segment_manager::wakeup() {
+    if (_message_queue.empty()) {
+        _message_queue.signal();
+    }
+}
+
+future<> db::commitlog::segment_manager::message_loop() {
+    bool shutdown = false;
+
+    while (!shutdown || !_message_queue.empty()) {
+        auto next_wakeup = no_timeout;
+        auto now = message_clock::now();
+
+
+        co_await _message_queue.wait(next_wakeup).handle_exception([](auto ignored) {});
```

Comment:
Let's avoid coroutine/continuation combos. Also, if you ignore an exception, need a comment to justify it. Usually it is a bug.

Reply:
It is ignoring timeouts. Since these are still generated and cause exceptions, we need to catch them. Catching them as a continuation is cheaper as it will not involve an actual throw (co_await, awaiter does throw comes _after_ timeout handling via future).